### PR TITLE
Add support for further customization of the call factory & switch the callback execution into another thread

### PR DESCRIPTION
**Pull request by kristinakoneva**
## Summary

- Added support for passing a `Call.Factory` which will be used by Halley to execute the requests (increasing the flexibility in configuring the `Call.Factory` instead of limiting it to an instance of `OkHttpClient`).
- Added an `AsyncCallFactory` wrapper for the call factory - the end goal is to avoid dispatcher exhaustion by switching the callback execution into another thread and ensure proper cancellation management.
- Exposed a `Retrofit.Builder.withHalley(configuration, call factory)` extension function in `:halley-retrofit` for easier Halley integration which should also ensure Retrofit’s call factory and the call factory Halley will use are correctly configured. Lib clients are encouraged to use this method when using Halley with Retrofit to avoid potential misconfiguration issues, therefore, I’ve marked some methods as deprecated which we will remove in some future release.

Huge shout to @antunflas for suggesting the idea of how dispatcher exhaustion can be avoided and helping me add support for this. 🙌 

## Changes

- Added `AsyncCallFactory` which executes the callback in a new thread. By default, OkHttp executes the callback code on the dispatcher thread, meaning that enqueuing another set of calls in the callback can exhaust the dispatcher (default maximum is 5). To free the dispatcher thread as soon as possible,  `AsyncCallback` is used to switch the callback execution into another thread from a non-fixed-sized thread pool. By doing so, `RelationshipLoader` blocks that thread instead of the dispatcher thread while waiting for relationship calls to complete.
- Added `Retrofit.Builder.withHalley(configuration, callFactory)` extension function in `:halley-retrofit`. It is important that Retrofit uses a call factory instance wrapped with `AsyncCallFactory` when making the initial call and after receiving the response, Halley should make any additionally required requests using the same call factory which is also appropriately wrapped with `AsyncCallFactory`. This extension function ensures a properly wrapped call factory is set to Retrofit and Halley is added as a converter factory.
- Updated all the places where `OkHttpClient` was requested with `Call.Factory` to allow more flexibility in customizing the call factory.
- Adjusted the sample app to the new changes. I’ve run the app and I didn’t notice any issues. Please double check the changes in the Ktor module since I’m not that familiar with Ktor in general.
- Updated the `README.md`.

### Type

- [X] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [ ] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

> [!WARNING]
> There are no breaking changes for now, the compatibility is kept. However, lib clients are strongly encouraged to switch to using `Retrofit.Builder.withHalley(configuration, callFactory)` extension function instead of `Json.asHalleyConverterFactory(configuration, httpClient)` and `BinaryFormat.asHalleyConverterFactory(configuration, httpClient)` to avoid dispatcher exhaustion and ensure the call cancellation is handled correctly. These methods will be removed in future releases, so they have been marked as deprecated.

## Checklist

- [X] I have performed a self-review of my own code.
- [X] I have tested my changes, including edge cases.
- [X] I have added necessary tests for the changes introduced (if applicable).
- [X] I have updated the documentation to reflect my changes (if applicable).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -191,13 +191,26 @@ val actual: HalModel = halley.decodeFromString(
 Retrofit.Builder()
     .addCallAdapterFactory(RxJava3CallAdapterFactory.create()) // Optional only if you use RxJava
     // Halley for Retrofit provides an extension on KotlinX Serialization Json 
-    .addConverterFactory(Json.asHalleyConverterFactory())
+    .addConverterFactory(Json.asHalleyConverterFactory(callFactory = callFactory))
     .addConverterFactory(ScalarsConverterFactory.create())
-    .client(httpClient)
+    .callFactory(callFactory)
```
Comment (antunflas) [inline README.md]: Imo no need to rename this method, it's the same method, we just made the argument more abstract. OkHttpClient still fits into it, and generally speaking, call factory is a networking client 🤷

File: README.md [hunk 2]
```diff
@@ -191,13 +191,26 @@ val actual: HalModel = halley.decodeFromString(
...
+> By default, OkHttp executes the callback code on the dispatcher thread, meaning that enqueuing another set of calls in the callback can
+> exhaust the dispatcher (default maximum is 5). To free the dispatcher thread as soon as possible, use this factory to **switch the callback
+> execution into another thread from a non-fixed-sized thread pool**. By doing so, that thread (instead of the dispatcher thread) is blocked
+> while waiting for relationship calls to complete.
+> 
+> _Example of creating such `Call.Factory` instance:_
+> ```kotlin
+> val callFactory = OkHttpClient.Builder().build().asyncCallFactory()
+> ```
+
```
Comment (antunflas) [inline README.md]: I'd remove this from the readme. This should be handled from the library itself. Clients can pass whatever they want.

File: halley-core/src/main/kotlin/com/infinum/halley/core/Halley.kt [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
 import kotlin.reflect.KClass
 import kotlinx.serialization.json.Json
 import kotlinx.serialization.serializer
+import okhttp3.Call
 import okhttp3.OkHttpClient
 
 public class Halley(
     private val configuration: Configuration = Configuration(),
-    httpClient: OkHttpClient = OkHttpClient.Builder().build()
+    callFactory: Call.Factory = OkHttpClient.Builder().build().asyncCallFactory(),
```
Comment (antunflas) [inline halley-core/src/main/kotlin/com/infinum/halley/core/Halley.kt]: Library should wrap into `asyncCallFactory` for any `callFactory`, not just in the default case.

File: halley-core/src/main/kotlin/com/infinum/halley/core/Halley.kt [hunk 4]
```diff
@@ -56,7 +58,7 @@ public class Halley(
     }
 
     init {
-        HttpClientCache.save(httpClient)
+        CallFactoryCache.save(callFactory)
```
Comment (antunflas) [inline halley-core/src/main/kotlin/com/infinum/halley/core/Halley.kt]: Meaning, this might be a good place to wrap the client so everywhere we have the wrapped CF.
```suggestion
        CallFactoryCache.save(callFactory.asyncCallFactory())
```

File: halley-core/src/main/kotlin/com/infinum/halley/core/extensions/CallFactory.kt [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * meaning that enqueuing another set of calls in the callback can exhaust the dispatcher (default maximum is 5).
+ * To free the dispatcher thread as soon as possible,
+ * [AsyncCallback] is used to switch the callback execution into another thread from a non-fixed-sized thread pool.
+ * By doing so, [RelationshipLoader] blocks that thread instead of the dispatcher thread
+ * while waiting for relationship calls to complete.
+ *
+ * @param executorService the executor service to use
+ * @return a new [Call.Factory] that executes the callback in a new thread
+ */
+public fun Call.Factory.asyncCallFactory(
```
Comment (antunflas) [inline halley-core/src/main/kotlin/com/infinum/halley/core/extensions/CallFactory.kt]: This should be internal

File: halley-ktor/src/main/java/com/infinum/halley/ktor/extensions/Configuration.kt [hunk 6]
```diff
@@ -38,7 +40,7 @@ public fun Configuration.defaultConfiguration(
     allowSpecialFloatingPointValues: Boolean =
         Json.Default.configuration.allowSpecialFloatingPointValues,
     useAlternativeNames: Boolean = Json.Default.configuration.useAlternativeNames,
-    httpClient: OkHttpClient = OkHttpClient.Builder().build()
+    callFactory: Call.Factory = OkHttpClient.Builder().build().asyncCallFactory(),
```
Comment (antunflas) [inline halley-ktor/src/main/java/com/infinum/halley/ktor/extensions/Configuration.kt]: This asyncCallFactory to the default client is not needed here since we just pass it to the Halley and there we will wrap.

Comment (antunflas) [inline halley-ktor/src/main/java/com/infinum/halley/ktor/extensions/Configuration.kt]: Perhaps we could keep the naming as `httpClient` to keep the compatibility.

File: halley-core/src/main/kotlin/com/infinum/halley/core/loader/HalRelationshipLoader.kt [hunk 7]
```diff
@@ -20,7 +20,7 @@ internal class HalRelationshipLoader : RelationshipLoader {
 
             val requestUrl = appendParameters(url, request.name, request.options)
 
-            HttpClientCache.load().newCall(
+            CallFactoryCache.load().newCall(
```
Comment (antunflas) [inline halley-core/src/main/kotlin/com/infinum/halley/core/loader/HalRelationshipLoader.kt]: In this class, we should also catch the `InterruptedException` and cancel the calls being made (here one, below all of them)

File: halley-retrofit/src/main/kotlin/com/infinum/halley/retrofit/extensions/Json.kt [hunk 8]
```diff
@@ -1,28 +1,30 @@
...
         configuration = configuration,
-        httpClient = httpClient
+        callFactory = callFactory
     )
 
 @JvmName("create")
 public fun Json.asHalleyConverterFactory(
     configuration: Halley.Configuration = Halley.Configuration(),
-    httpClient: OkHttpClient = OkHttpClient.Builder().build()
+    callFactory: Call.Factory = OkHttpClient.Builder().build().asyncCallFactory(),
```
Comment (antunflas) [inline halley-retrofit/src/main/kotlin/com/infinum/halley/retrofit/extensions/Json.kt]: Redundant here as well, Halley will do the wrapping.

File: sample/src/main/kotlin/com/infinum/halley/sample/mock/server/SampleWebServer.kt [hunk 9]
```diff
@@ -17,13 +19,13 @@ class SampleWebServer {
...
 
-    private val okhttp: OkHttpClient =
+    private val callFactory: Call.Factory =
         OkHttpClient.Builder()
             .addInterceptor(
                 HttpLoggingInterceptor()
                     .apply { level = HttpLoggingInterceptor.Level.BODY }
             )
-            .build()
+            .build().asyncCallFactory()
```
Comment (antunflas) [inline sample/src/main/kotlin/com/infinum/halley/sample/mock/server/SampleWebServer.kt]: This should be removed then

File: sample/src/main/kotlin/com/infinum/halley/sample/ui/MainActivity.kt [hunk 10]
```diff
@@ -117,14 +118,14 @@ class MainActivity : AppCompatActivity() {
...
-            httpClient = OkHttpClient.Builder()
+            callFactory = OkHttpClient.Builder()
                 .addInterceptor(
                     HttpLoggingInterceptor()
                         .apply {
                             level = HttpLoggingInterceptor.Level.BODY
                         }
                 )
-                .build()
+                .build().asyncCallFactory()
```
Comment (antunflas) [inline sample/src/main/kotlin/com/infinum/halley/sample/ui/MainActivity.kt]: And in this class as well

**Review by antunflas — COMMENTED** [review]
Good job 👍 
Please check my comments, I think we can keep `asyncCallFactory` internal so clients don't have to care about it.

File: halley-ktor/src/main/java/com/infinum/halley/ktor/extensions/Configuration.kt (hunk 6 above)
Comment (KCeh) [reply halley-ktor/src/main/java/com/infinum/halley/ktor/extensions/Configuration.kt]: Since we are not X.0.0 version I think we can get away with breaking changes to public API. But, for this case let's follow Antun's advice.

But just keep in mind when there are breaking changes in public API we should bump major version (in major.minor.patch format) 
You can also use https://github.com/Kotlin/binary-compatibility-validator to keep track of public API

File: halley-core/src/main/kotlin/com/infinum/halley/core/extensions/CallFactory.kt [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    executorService: ExecutorService = Executors.newCachedThreadPool(),
```
Comment (KCeh) [inline halley-core/src/main/kotlin/com/infinum/halley/core/extensions/CallFactory.kt:24]: Could you explain the motivation/problem behind these changes in more detail?

> `the end goal is to avoid dispatcher exhaustion by switching the callback execution into another thread`

Can't we use our dispatcher with a predefined thread pool size?
`val dispatcher = Executors.newFixedThreadPool(128).asCoroutineDispatcher()`

What are the downsides of using `Executors.newCachedThreadPool()`?
I see that cached thread pool **works best when we’re dealing with a reasonable number of short-lived tasks** 
This is our use case?

Creating threads (and cleaning threads) on demand can be demanding (pun intended) on the system.
What are the worst case scenarios we are expecting?
If we reach scenario of 100+ threads in pool that could cause system performance issues due to context switching. Have we thought about that?

I am no expert in thread pool size optimizations or even HTTP clients. 
I am just trying to get to the bottom of the issue and see why we implemented improvement like this. Changes look ok 👍

**Comment by kristinakoneva** [comment]
After some discussions, we have decided to make additional changes to achieve proper configuration and easier integration.

It is important that both the initial call Retrofit’s call factory makes and all the additional ones Halley makes after deserializing the response body are executed with a call factory instance wrapped with `AsyncCallFactory`.

The intial (first) call is executed by the call factory / client which is set when configuring Retrofit. Halley being a converter factory, takes over once it has a response body it needs to convert. When Halley detects embedded properties, it uses the call factory which was passed to it when it was added as converter factory to make the additional requests after the initial call.

Therefore, doing the wrapping internally to the call factory passed to Halley is not enough - the initial call will still be executed with the call factory set to Retrofit and if that call factory is not wrapped with `AsyncCallFactory`, the rest of the logic won’t work inside of Halley. To avoid misconfiguration issues and putting the burden of setting a wrapped instance of the call factory on the lib client (integrator), we will add a `Retrofit.Builder.withHalley(configuration, callFactory)` extension function.

This function will take the responsibility of:
- setting the call factory used by Retrofit for the initial (first) call (the library will do the wrapping to `AsyncCallFactory` here in the Retrofit builder when calling `.callFactory`)
- adding Halley as a converter factory (inside the Halley class, the library will do the wrapping to `AsyncCallFactory` and ensure Halley uses a wrapped instance for all the embedded requests it will potentially make).

Call factories other than OkHttpClient might not have these exact problems, however, we decided to do the wrapping for all regardless so that Halley takes control of managing this stuff and lib clients don’t have to worry about it.

I’ve updated the implementation and the PR description accordingly.

> [!NOTE]
> I kept `asyncCallFactory` extension function public for now because I had to use it the Retrofit module, but it is defined in the core one. Tried to annotate it with something like `RestrictTo` from AndroidX annotation library to make sure it is visible only for the library group, but either I didn’t do it correctly or it wasn’t working cause of the module ids setup 😅, not sure. Let me know if there is some other way to expose `asyncCallFactory` only for the library modules and not the lib clients as well. Maybe it won’t hurt leaving it public for now while the other deprecated methods are still present 🤔, so that lib clients can potentially do the wrapping themselves, but I don’t know, we do not recommend this for the long run.

File: halley-core/src/main/kotlin/com/infinum/halley/core/extensions/CallFactory.kt (hunk 11 above)
Comment (kristinakoneva) [reply halley-core/src/main/kotlin/com/infinum/halley/core/extensions/CallFactory.kt:24]: I’ll try to explain the motivation the best I can and after that maybe we can figure out the best optimization together 🙏  cause I’m no expert in this more than you are. :sweat_smile:

The OkHttpClient dispatcher exhaustion issue was noticed when the lib client canceled a certain initial Retrofit call, 5 embedded calls were already enqueued which don’t pick up the cancellation nor execute and the lib client ended up in some frozen state (max number of enqueued requests was reached - default is 5 - can be changed to a higher number by just changing OkHttpClient’s max request limit, but regardless, that doesn’t solve the cancellation issue - all of the related calls to the callback should be cancelled).

We don’t want to have this call limitation and we want control over the cancellation to make sure all embedded calls are canceled when the initial Retrofit call is cancelled by the client (lib client can trigger cancellation only on the initial call, it doesn’t have reference to the rest which Halley handles). We can achieve this by switching the callback execution into another thread which we can manage.
By creating a delegate of the call and wrapping it in our `AsyncCall`, we have control over the `Call#cancel` method and by overriding it and using our own executor service to execute the requests, we can make sure to cancel all the other calls by interrupting the callback in `AsyncCall#cancel`. Previously, when a completed call was cancelled, Retrofit would trigger cancel and forward it to OkHttp, but OkHttp wouldn’t fire the callback since that initial request was already completed, so there was no way for us to pick up the cancellation and propagate it to the other embedded requests. 

Btw, what I refer to “initial call / request” is the first request that is made before following all the embedded ones from the defined models. The call factory / client lib clients have set for Retrofit, executes this initial call. Halley then makes additional requests if needed (if the response body has some embedded properties) and it uses the call factory lib clients have passed to it.

**TL;DR** The main motivation is avoiding OkHttpClient’s dispatcher exhaustion by not fixing the max enqueued call size along with more control over the cancellation management. 

Let me know if something is still unclear, I’m not sure I explained the cancellation management the best and that’s the part which makes a difference here. We don’t want to simply avoid the exhaustion by increasing the default number of requests OkHttpClient can make at one time (we can easily do that by passing a dispatcher in the OkHttpClient builder with a high number set for the max requests & max requests per host properties), but rather, make sure `HalRelationshipLoader` blocks a thread we’ve intentionally switched the callback execution to.

File: halley-ktor/src/main/java/com/infinum/halley/ktor/extensions/Configuration.kt (hunk 6 above)
Comment (kristinakoneva) [reply halley-ktor/src/main/java/com/infinum/halley/ktor/extensions/Configuration.kt]: With the latest changes, I kept the compatibility and marked some methods as deprecated, so we won’t have any breaking changes, although it is recommended that lib clients migrate to the new setup.

Regarding the versioning, maybe it would be a good idea to have a 1.0.0 release now since the library is no longer in alpha/beta mode.

Once we agree what the next version will be, I’ll create a release branch and rebase this PR.

File: halley-core/src/main/kotlin/com/infinum/halley/core/loader/HalRelationshipLoader.kt (hunk 7 above)
Comment (kristinakoneva) [reply halley-core/src/main/kotlin/com/infinum/halley/core/loader/HalRelationshipLoader.kt]: [Updated](https://github.com/infinum/android-halley/pull/8/commits/affca891c29992c4c6d549d3ecd5c99449b970c1). Please double check if it has been caught in the correct places. 🙏 

Btw, I still wasn’t able to catch any `InterruptedException`s and yet, I couldn’t see logs for the other requests (if they were potentially never actually interrupted/cancelled). 🤔 

I also tried using exactly the same wrapped instance of the call factory - the same executor service would be used this way I think (updated the extension function as it is below and removed the wrapping when saving the call factory in Halley):
```
public fun Retrofit.Builder.withHalley(
    configuration: Halley.Configuration = Halley.Configuration(),
    callFactory: Call.Factory = OkHttpClient.Builder().build(),
): Retrofit.Builder  {
    val callFactoryy = callFactory.asyncCallFactory()
    return this.callFactory(callFactoryy)
        .addConverterFactory(HalleyConverterFactory(configuration, callFactoryy))
}
```
to see if that would make any difference, but it didn’t as far as I could see.

File: halley-core/src/main/kotlin/com/infinum/halley/core/extensions/CallFactory.kt (hunk 11 above)
Comment (KCeh) [reply halley-core/src/main/kotlin/com/infinum/halley/core/extensions/CallFactory.kt:24]: Great, thanks for the explanation. I didn't realise cancellation was the biggest motivation behind changes.
Also, good work 👏 

My only comment would be to consider if we can reach the limits of the thread pool and what will happen then?
Or maybe we can write some synthetic stress test or even integration test?

**Comment by KCeh** [comment]
> I kept `asyncCallFactory` extension function public for now because I had to use it the Retrofit module, but it is defined in the core one. Tried to annotate it with something like `RestrictTo` from AndroidX annotation library to make sure it is visible only for the library group, but either I didn’t do it correctly or it wasn’t working cause of the module ids setup 😅, not sure. Let me know if there is some other way to expose `asyncCallFactory` only for the library modules and not the lib clients as well. Maybe it won’t hurt leaving it public for now while the other deprecated methods are still present 🤔, so that lib clients can potentially do the wrapping themselves, but I don’t know, we do not recommend this for the long run.

I might be missing something. But, `RestrictTo` seems promising, try to define `group` by using `releaseConfig.group` in `build.gradle` file -> `group = releaseConfig.group`
I think right now it is only define for publishing

File: halley-core/src/main/kotlin/com/infinum/halley/core/loader/HalRelationshipLoader.kt [hunk 12]
```diff
@@ -20,11 +21,16 @@ internal class HalRelationshipLoader : RelationshipLoader {
...
                 Request.Builder()
                     .url(requestUrl)
                     .build()
-            ).enqueue(callback)
+            )
+            try {
+                call.enqueue(callback)
+            } catch (ex: InterruptedException) {
+                call.cancel()
+            }
```
Comment (antunflas) [inline halley-core/src/main/kotlin/com/infinum/halley/core/loader/HalRelationshipLoader.kt]: It's not good; we need to try-catch the `await` part (as well). Since await is the blocking function, enqueue is async, so it completes immediately.

File: halley-core/src/main/kotlin/com/infinum/halley/core/loader/HalRelationshipLoader.kt [hunk 13]
```diff
@@ -40,22 +46,30 @@ internal class HalRelationshipLoader : RelationshipLoader {
...
-            }
-            .forEach { CallFactoryCache.load().newCall(it).enqueue(callback) }
+                .forEach {
+                    val newCall = CallFactoryCache.load().newCall(it)
+                    calls.add(newCall)
+                    newCall.enqueue(callback)
+                }
+        } catch (ex: InterruptedException) {
+            calls.forEach { it.cancel() }
+        }
```
Comment (antunflas) [inline halley-core/src/main/kotlin/com/infinum/halley/core/loader/HalRelationshipLoader.kt:69]: Same goes here

```
val calls = requests
    .requests
    .mapNotNull {
        replaceTemplatePlaceholders(it)?.let { url ->
            val requestUrl = appendParameters(url, it.name, it.options)
            Request.Builder()
                .url(requestUrl)
                .build()
         }
    }
    .map {
        CallFactoryCache.load().newCall(it)
    }

try {
    calls.forEach { it.enqueue(callback) }
    countDownLatch.await()
} catch (interrupt: InterruptedException) {
    calls.forEach { it.cancel() }
}
```

File: halley-core/src/main/kotlin/com/infinum/halley/core/extensions/CallFactory.kt (hunk 11 above)
Comment (antunflas) [reply halley-core/src/main/kotlin/com/infinum/halley/core/extensions/CallFactory.kt:24]: > if we can reach the limits of the thread pool and what will happen then?

CachedThreadPool has no thread limit; the only limit is the system resources.

File: halley-core/src/main/kotlin/com/infinum/halley/core/loader/HalRelationshipLoader.kt (hunk 7 above)
Comment (antunflas) [reply halley-core/src/main/kotlin/com/infinum/halley/core/loader/HalRelationshipLoader.kt]: I don't see this change in the list

File: halley-core/src/main/kotlin/com/infinum/halley/core/extensions/CallFactory.kt (hunk 5 above)
Comment (antunflas) [reply halley-core/src/main/kotlin/com/infinum/halley/core/extensions/CallFactory.kt]: If it won't be internal, let's just name it less generic. Something like `halleyCallFactory` might be good.

File: halley-core/src/main/kotlin/com/infinum/halley/core/loader/HalRelationshipLoader.kt (hunk 12 above)
Comment (kristinakoneva) [reply halley-core/src/main/kotlin/com/infinum/halley/core/loader/HalRelationshipLoader.kt]: Ahh right, my bad, I’ll fix it, thanks!

File: halley-core/src/main/kotlin/com/infinum/halley/core/loader/HalRelationshipLoader.kt [hunk 14]
```diff
@@ -28,8 +28,9 @@ internal class HalRelationshipLoader : RelationshipLoader {
             try {
                 call.enqueue(callback)
                 countDownLatch.await()
-            } catch (ex: InterruptedException) {
+            } catch (interruptedException: InterruptedException) {
                 call.cancel()
+                throw interruptedException
```
Comment (kristinakoneva) [inline halley-core/src/main/kotlin/com/infinum/halley/core/loader/HalRelationshipLoader.kt]: We are rethrowing the exception because otherwise the cancellation doesn’t propagate to the other calls.

**Review by antunflas — APPROVED** [review]
👍

**Comment by kristinakoneva** [comment]
**Update on the visibility of the `asyncCallFactory` extension function topic:**

The AndroidX Annotation lib actually worked fine without any additional setup - I probably missed sth the first time. Annotating the function with `RestrictTo` resulted in a lint warning on the lib client side. However, in order to use the annotation, we have to add the AndroidX Annotation dependency which might lead to dependency conflicts for existing lib clients, therefore, we've decided not to include it to avoid unnecessary integration issues. Lib clients would have been able to suppress and ignore the lint error anyways, so we wouldn't have been able to prevent usage nevertheless.

I've [renamed the function](https://github.com/infinum/android-halley/pull/8/commits/ce4308040b4eccac67483cc6aadc178f8f596069) as suggested in [this comment](https://github.com/infinum/android-halley/pull/8#discussion_r1910315905).
